### PR TITLE
feat: Issue #2 - Comprehensive Pydantic Data Models

### LINKED ISSUE #2: 📊 Data Models: Create Pydantic models for articles and analysis results
## 📋 Task Description
Create core data models using Pydantic for representing news articles, analysis results, and system entities.

## 🎯 Acceptance Criteria
- [ ] Create `models/` directory in the project
- [ ] Define `Article` model with fields:
  - `id`, `title`, `content`, `url`, `source`, `published_at`, `scraped_at`
  - `author`, `tags`, `summary`
- [ ] Define `AnalysisResult` model with fields:
  - `article_id`, `bias_score`, `credibility_score`, `sentiment`
  - `fact_check_claims`, `summary`, `analyzed_at`
- [ ] Define `NewsSource` model with fields:
  - `name`, `url`, `bias_rating`, `credibility_rating`, `active`
- [ ] Define `FactCheckClaim` model with fields:
  - `claim_text`, `verification_status`, `sources`, `confidence`
- [ ] Add validation rules and custom validators
- [ ] Create model mixins for common fields (timestamps, etc.)

## 🛠️ Technical Notes
- Use Pydantic v2 for all models
- Add proper type hints and docstrings
- Include field validation and constraints
- Consider using `datetime` for timestamps with timezone awareness

## 📚 References
- [Pydantic Documentation](https://docs.pydantic.dev/)

**Labels:** backend, models, medium-priority

### PR DESCRIPTION
**Pull request by MaxGaipl**
## 📊 Issue #2: Pydantic Data Models

This PR implements comprehensive Pydantic data models for the AI News Analyst application, providing type-safe and validated data structures for all core entities.

### ✅ Completed Requirements

- **📁 Models Directory**: Complete implementation in `src/news_analyst/models/`
- **🏗️ Base Infrastructure**: 
  - `BaseNewsModel` with common Pydantic configuration
  - `TimestampMixin` for created_at/updated_at fields
  - `UUIDMixin` for UUID primary keys
  
- **📰 Article Model**: 
  - Core fields: `id`, `title`, `content`, `url`, `source`, `published_at`, `scraped_at`
  - Extended fields: `author`, `tags`, `summary`, `language`, `word_count`
  - Auto word count calculation and tag deduplication validation

- **📺 NewsSource Model**:
  - Fields: `name`, `url`, `bias_rating`, `credibility_rating`, `active`
  - Extended: `country`, `language`, `description`, `rss_feeds`, `scraping_selectors`
  - Bias and credibility rating enums with helper methods

- **✅ FactCheckClaim Model**:
  - Fields: `claim_text`, `verification_status`, `sources`, `confidence`
  - Extended: `context`, `verification_notes`
  - Verification status enum and credibility helper methods

- **📈 AnalysisResult Model**:
  - Core fields: `article_id`, `bias_score`, `credibility_score`, `sentiment`
  - Analysis fields: `fact_check_claims`, `summary`, `analyzed_at`
  - Performance: `analysis_version`, `processing_time_ms`, `confidence_score`
  - Helper methods for bias/credibility labels and claim analysis

### 🛠️ Technical Implementation

- **✨ Pydantic v2**: Modern validation with `BaseModel` and `ConfigDict`
- **🔒 Type Safety**: Comprehensive type hints and field constraints
- **📋 Validation**: Custom validators for tags, sources, scores
- **📊 Enums**: Structured enums for bias, credibility, verification status, sentiment
- **🧪 Comprehensive Tests**: 9 test cases covering all models and edge cases
- **📖 Documentation**: Detailed docstrings and field descriptions

### 🧪 Validation

```bash
# All tests pass
uv run pytest tests/unit/test_models.py -v
# 9 passed in 0.13s

# Verification script demonstrates functionality
uv run python scripts/verify_issue2.py
# ✅ All models work correctly
```

### 🏗️ Model Architecture

```
BaseNewsModel (common config)
├── UUIDMixin + TimestampMixin (base functionality)
├── Article (news article data)
├── NewsSource (media outlet info)
├── FactCheckClaim (individual claims)
├── AnalysisResult (complete analysis)
└── Sentiment (sentiment analysis results)
```

### 📚 Public API

Clean module exports via `__all__`:
- Base models and mixins for extension
- Core business models for data representation  
- Enums for type safety and validation

### 🔄 Next Steps

After merging this PR:
- **Issue #3**: Web Scraping Engine (Crawl4AI) 
- **Issue #4**: LLM Analysis Agents (Pydantic AI)
- **Issue #5**: Database Layer (SQLAlchemy)

---
**Closes #2**

### CODE REVIEW

File: src/news_analyst/models/base.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+from datetime import datetime
+from typing import Optional
+from uuid import UUID, uuid4
+
+from pydantic import BaseModel, Field, ConfigDict
+
+
+class TimestampMixin(BaseModel):
```
Comment (MaxGaipl) [inline src/news_analyst/models/base.py:12]: Can you explain the use of Mixins here? Is this just used to not reduce code duplication?

File: src/news_analyst/models/fact_check.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    TRUE = "true"
+    FALSE = "false"
+    PARTIALLY_TRUE = "partially_true"
+    MISLEADING = "misleading"
+    UNVERIFIED = "unverified"
+    DISPUTED = "disputed"
+
+
+class FactCheckClaim(BaseNewsModel, UUIDMixin, TimestampMixin):
```
Comment (MaxGaipl) [inline src/news_analyst/models/fact_check.py:25]: From what I know, multiple inheritance is not a clean code technique. Can you explain, why it is used here?

File: src/news_analyst/models/analysis.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        """Validate fact check claims."""
+        return v if v else []
+
+    @field_validator("bias_score", "credibility_score")
+    @classmethod
+    def validate_scores(cls, v: float) -> float:
+        """Ensure scores are within valid range."""
+        return max(0.0, min(1.0, v))
+
+    def get_bias_label(self) -> str:
```
Comment (MaxGaipl) [inline src/news_analyst/models/analysis.py:113]: why not use an enum for the return types to streamline the code?
